galaxy_ng/app/access_control/access_policy.py

```diff
@@ -91,6 +91,25 @@ def can_create_collection(self, request, view, permission):
             raise NotFound(_("Namespace in filename not found."))
         return request.user.has_perm("galaxy.upload_to_namespace", namespace)
 
+    def can_sign_collections(self, request, view, permission):
+        # Repository is required on the CollectionSign payload
+        # Assumed that if user can modify repo they can sign everything in it
```

Comment:
Does this imply a partner (not partner engineering) could upload a new collection and then sign it while it resides in the staging repo? Does that same signature carry over to the published repo if the collection is moved/certified? Is any of that a concern or am I thinking too hard about it?

Reply:
This permission is combined on https://github.com/ansible/galaxy_ng/pull/1145/files#diff-a4b60e9eac0488f8168423dfd36066e96232ef13e886edd13e96747bfae12984 so to be able to perform signature upload or call /sign/ the user must also have rh-entitlements, so only PEs

Reply:
Most all c.rh.c users have rh-entitlements, that is not an indicator of a RH Partner Engineering user. But we would only give  the `can_sign_collections` permission to a RH Partner Engineering user.

Reply:
@awcrosby notice that `can_sign_collections` is a dynamic permission, it is calculates based on `request.user.has_perm('ansible.modify_ansible_repo_content', repository)` and `request.user.has_perm('galaxy.upload_to_namespace', namespace)` So we assume that if user can add/edit repo and namespace the same user would be able to click the sign button, however that applies only for on-premise since on CRC we are not going to have a `sign` button.